### PR TITLE
Get units tests (continued) and assorted test cleanups/fixes

### PR DESCRIPTION
**Pull request by phlax**
- Improves handling of uids and initial params
- Adds tests(s) for non-initial get_units requests

### CODE REVIEW

File: pytest_pootle/env.py [hunk 1]
```diff
@@ -440,5 +409,18 @@ def _add_submissions(self, unit, created):
             unit.markfuzzy()
 
         # mark OBSOLETE
-        if original_state == OBSOLETE:
+        elif original_state == OBSOLETE:
             unit.makeobsolete()
+
+        elif unit.target:
+            # edit unit again
```
Comment (unho) [inline pytest_pootle/env.py]: So this is changing translation for already translated unit, isn't it?

Comment (phlax) [inline pytest_pootle/env.py]: yep

File: pytest_pootle/search.py [hunk 2]
```diff
@@ -23,18 +23,22 @@
...
 
 def calculate_search_results(kwargs, user):
-    initial = kwargs["initial"]
     pootle_path = kwargs["pootle_path"]
 
     category = kwargs.get("category")
     checks = kwargs.get("checks")
+    initial = kwargs.get("initial", False)
     limit = kwargs.get("count", 9)
     modified_since = kwargs.get("modified_since")
```
Comment (unho) [inline pytest_pootle/search.py]: Is it not necessary to alter this too to `modified-since`?

Comment (phlax) [inline pytest_pootle/search.py]: hmm - we could do it instead i think

Comment (phlax) [inline pytest_pootle/search.py]: and do the parsing here too, actually

File: pytest_pootle/env.py (hunk 1 above)
Comment (unho) [inline pytest_pootle/env.py]: Then I would rewrite this particular comment as it is not 100% clear to me.

**Comment by unho** [comment]
lgtm

**Comment by ta2-1** [comment]
gtm

**Pull request merged**